### ISSUE TITLE
why the "times" data was transformed by "torch.log" before being sent to the model

### ISSUE THREAD

**Issue by jingge326** [mbilos/neural-flows-experiments#2]
Dear author,

In file `experiments/tpp/experiment.py`, line 34, the "times" data was transformed by a `log `function. 

`times = torch.log(times + 1)`

Would you like to give some insights on this? 

Based on my knowledge, the `log `transformation itself would change the data distribution. For example, If the data follows a log-normal distribution. After the `log `transformation, it would follow a normal distribution. Would the transformation affect the model's performance? Can we drop the `log `transformation in other applications?

**Comment by mbilos** [comment]
We use log-transform since time values can be large which can lead to numerical/training issues. You can instead normalize time before, or omit this line completely.

You are correct, this will change the data distribution. However, you can treat log as a random variable transformation and apply the [change of variables](https://en.wikipedia.org/wiki/Probability_density_function#Function_of_random_variables_and_change_of_variables_in_the_probability_density_function) which will give you the correct likelihood in the end. That is, you "won't notice" changing the distribution as you will account for the transformation in the likelihood. We do exactly that in the few lines below when calculating the loss. You can take a look at normalizing flows in general and our other paper that does something similar: https://github.com/shchur/ifl-tpp.

**Comment by won-bae** [comment]
Thank you for the explanation. It now makes sense why there is the addition of `times` in https://github.com/mbilos/neural-flows-experiments/blob/bd19f7c92461e83521e268c1a235ef845a3dd963/nfe/experiments/tpp/experiment.py#L40
